added functionality for bookUpdate

gcater opened the pull request:
The landing page index.tsx now only has a bookForm component. On a successful book create the user will get sent to a new page "src/pages/book/[bookid]/index.tsx". This page has a BookUpdate Component and the BookRenderer component. User can make changes to the book info or add a chapter.

book.ts got updateBook mutation and getBook query. BookRenderer component now uses getBook which requires a bookId.

src/pages/book/[bookid]/addChapter.tsx and src/pages/book/[bookid]/[chapterid]/addSection.tsx now send BookRenderer a bookId var.

Evanc123 commented on src/components/BookUpdate.tsx:
```diff
@@ -0,0 +1,103 @@
+});
+
+const BookUpdate = ({ bookId }: { bookId: string }): JSX.Element => {
+  const { data: queryData, refetch: refetchBook } =
```
Rename

gcater replied:
Is this because there are other functions basically named the same thing?

Evanc123 replied:
Query data is just unclear

Evanc123 commented on src/components/BookUpdate.tsx:
```diff
@@ -0,0 +1,103 @@
+  });
+  const { mutate: deleteBook } = api.book.deleteBook.useMutation({
+    onSuccess: () => {
+      window.location.href = "/";
```
Fix

Evanc123 commented on src/components/BookUpdate.tsx:
```diff
@@ -0,0 +1,103 @@
+          <>
+            <CardContent>
+              <Button>
+                <a href={`/book/${bookId}/addChapter`}>Add Chapter</a>
```
Never use a href in a next application, always use next link <Link />

gcater replied:
<a onClick={() => router.push(`/book/${bookId}/addChapter`)}>
 
 Would this be a correct alternative?

Evanc123 replied:
That would be okay, but next link is better

Evanc123 replied:
https://nextjs.org/docs/pages/api-reference/components/link 

This is the primary way, but if hou need to do it programmatically you y can use router push (but in this case it's just a button)

gcater commented:
Changed filename UpdateBook.tsx to EditBook.tsx. No longer using href, instead using router.push.

gcater commented:
Changed EditBook back to BookUpdate, renamed "querydata" to "updated book", changed the "a hrefs" to "link hrefs" in BookForm and ChapterForm. Made the BookForm take up the whole screen in pages/index.tsx

Evanc123 commented on src/components/BookUpdate.tsx:
```diff
@@ -0,0 +1,106 @@
+});
+
+const BookUpdate = ({ bookId }: { bookId: string }): JSX.Element => {
+  const { data: updatedBook, refetch: refetchBook } =
```
yo ucan just call it `book`

Evanc123 commented on src/pages/index.tsx:
```diff
@@ -18,11 +17,7 @@ export default function Home() {
-          <div className="w-1/2">{BookForm()}</div>
-          <div className="w-1/2">{BookRenderer()}</div>
-        </div>
+        <div className="w-1/2">{BookForm()}</div>
```
i'd rather have this be a component than a function, but we can address later
